src/minType.js

```diff
@@ -133,7 +131,7 @@ function minType (sup, sub) {
   // 2. if `super-type` and `sub-type` have the same value and the value is in the set `any boolean datetime datetime-only number integer string null file xml json"`
   if (superType === subType && isOpaqueType(superType)) {
     // 2.1. we initialize the variable `computed` to the record with property `type` having the common `super-type` and `sub-type` value
-    const computed = _.cloneDeep(sup)
+    const computed = Object.assign({}, sup, sub)
```

Comment:
I was intentionally doing a deep copy here, do you think a shallow copy is fine?

Reply:
Deep copies weren't done consistently in `minType`, so I assumed they were being done conservatively. `sub` never seems to be cloned, presumably because `canonicalForm` creates the `sub` object. On the other hand, `sup` is deep-cloned for case 2, but not for case 8, and only conditionally for case 3. So for `sup`, I took the approach that a shallow copy was okay because `minType` never modifies nested objects or arrays. Particularly, after this PR, I don't think `minType` assumes it can modify any given object directly, so I *think* shallow copies are okay. I don't think there's any internal aliasing, but I don't know what the external contract is. Please let me know if I'm missing a use case or if we just want to strictly avoid any aliasing.